verde/utils.py

```diff
@@ -201,3 +202,93 @@ def maxabs(*args):
     arrays = [np.atleast_1d(i) for i in args]
     absolute = [np.abs([i.min(), i.max()]).max() for i in arrays]
     return np.max(absolute)
+
+
+def grid_to_table(grid):
+    """
+    Convert a grid to a table with the values and coordinates of each point.
+
+    Takes a 2D grid as input, extracts the coordinates and runs them through
+    :func:`numpy.meshgrid` to create a 2D table. Works for 2D grids and
+    n-dimensional variables. Use cases includes passing gridded data to
+    functions that expect data in XYZ format such as :class:`verde.BlockReduce`
+
+    Parameters
+    ----------
+    grid : :class:`xarray.Dataset`
+        A 2D grid with a single data variable.
+
+    Returns
+    -------
+    table : :class:`pandas.DataFrame`
+        :class:`pandas.DataFrame` with coordinates and variables.
+
+    Examples
+    --------
+    >>> var1 = xr.DataArray(np.arange(20).reshape(4,5), coords = (np.arange(4), np.arange(5)))
+    >>> var2 = xr.DataArray(np.arange(20,40).reshape(4,5), coords = (np.arange(4), np.arange(5)))
+    >>> print(var1)
+    <xarray.DataArray (dim_0: 4, dim_1: 5)>
+    array([[ 0,  1,  2,  3,  4],
+           [ 5,  6,  7,  8,  9],
+           [10, 11, 12, 13, 14],
+           [15, 16, 17, 18, 19]])
+    Coordinates:
+      * dim_0    (dim_0) int32 0 1 2 3
+      * dim_1    (dim_1) int32 0 1 2 3 4
+
+    >>> ds = var1.to_dataset(name = 'var1')
+    >>> ds['var2'] = var2
+    >>> print(ds)
+    <xarray.Dataset>
+    Dimensions:  (dim_0: 4, dim_1: 5)
+    Coordinates:
+      * dim_0    (dim_0) int32 0 1 2 3
+      * dim_1    (dim_1) int32 0 1 2 3 4
+    Data variables:
+        var1     (dim_0, dim_1) int32 0 1 2 3 4 5 6 7 8 ... 12 13 14 15 16 17 18 19
+        var2     (dim_0, dim_1) int32 20 21 22 23 24 25 26 ... 33 34 35 36 37 38 39
+
+    >>> print(grid_to_table(ds))
+        dim_0  dim_1  var1  var2
+    0       0      0     0    20
+    1       1      0     1    21
+    2       2      0     2    22
+    3       3      0     3    23
+    4       0      1     4    24
+    5       1      1     5    25
+    6       2      1     6    26
+    7       3      1     7    27
+    8       0      2     8    28
+    9       1      2     9    29
+    10      2      2    10    30
+    11      3      2    11    31
+    12      0      3    12    32
+    13      1      3    13    33
+    14      2      3    14    34
+    15      3      3    15    35
+    16      0      4    16    36
+    17      1      4    17    37
+    18      2      4    18    38
+    19      3      4    19    39
```

Comment:
There seems to be something wrong with the generation of the coordinates. The value at `[3, 0]` should be 15 from the `print(var1)`. From https://github.com/fatiando/verde/blob/master/verde/coordinates.py#L433 the coordinate order in the meshgrid should be `dim_1, dim_0` instead. This is actually a good example of why using more readable variable names in the tests makes it easier to reason about the code.